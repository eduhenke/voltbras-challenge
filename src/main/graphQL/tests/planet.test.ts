import { createTestClient } from 'apollo-server-testing'
import ArcsecondApi from '../../../infra/db/arcsecond/helpers/arcsecond-api'
import { typeDefs, resolvers } from '../'
import { ApolloServer } from 'apollo-server'

describe('Planet', () => {
  test('Should return suitable planets', async () => {
    const arcsecondApi = new ArcsecondApi()

    const server = new ApolloServer({
      typeDefs,
      resolvers,
      dataSources: () => {
        return {
          arcsecondApi
        }
      }
    })

    const fakePlanetsQuery = {
      count: 0,
      next: 'any_value',
      previous: 'any_value',
      results: [
        {
          name: 'any_name',
          mass: {
            unit: 'any_unit',
            value: 12
          }
        },
        {
          name: 'other_name',
          mass: {
            unit: 'any_unit',
            value: 32
          }
        }
      ]
    }

    const SUITABLE_PLANETS_QUERY = `
        query {
            suitablePlanets {
                name,
                mass
            }
        }
    `

    arcsecondApi.getRequest = jest.fn(async () => fakePlanetsQuery)

    const { query } = createTestClient(server)

    const res = await query({ query: SUITABLE_PLANETS_QUERY })
    expect(res).toMatchSnapshot()
    expect(res.data.suitablePlanets.length).toBe(2)
  })
})
