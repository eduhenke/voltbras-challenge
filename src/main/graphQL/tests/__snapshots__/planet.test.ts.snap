// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet Should return suitable planets 1`] = `
Object {
  "data": Object {
    "suitablePlanets": Array [
      Object {
        "hasStation": false,
        "mass": 32,
        "name": "other_name",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
